ReactWindows/ReactNative.Net46/Touch/TouchHandler.cs

```diff
@@ -0,0 +1,358 @@
+
+                var pointerIndex = _pointers.Count;
+                _pointers.Add(pointer);
+                DispatchTouchEvent(TouchEventType.Start, _pointers, pointerIndex);
+            }
+        }
+
+        private void OnPointerMoved(object sender, MouseEventArgs e)
+        {
+            var pointerIndex = 1;
+            if (pointerIndex != -1)
+            {
+                var pointer = _pointers[pointerIndex];
+                UpdatePointerForEvent(pointer, e);
+                DispatchTouchEvent(TouchEventType.Move, _pointers, pointerIndex);
+            }
+        }
+
+        private void OnPointerReleased(object sender, MouseButtonEventArgs e)
+        {
+            OnPointerConcluded(TouchEventType.End, e);
+        }
+
+        private void OnPointerConcluded(TouchEventType touchEventType, MouseButtonEventArgs e)
+        {
+            var pointerIndex = 1;
+            if (pointerIndex != -1)
+            {
+                var pointer = _pointers[pointerIndex];
+                UpdatePointerForEvent(pointer, e);
+                DispatchTouchEvent(touchEventType, _pointers, pointerIndex);
+
+                _pointers.RemoveAt(pointerIndex);
+
+                if (_pointers.Count == 0)
+                {
+                    _pointerIDs = 0;
+                }
+
+                _view.ReleaseMouseCapture();
+            }
+        }
+
+        private int IndexOfPointerWithId(uint pointerId)
+        {
+            for (var i = 0; i < _pointers.Count; ++i)
+            {
+                if (_pointers[i].PointerId == pointerId)
+                {
+                    return i;
+                }
+            }
+
+            return -1;
+        }
+
+        private UIElement GetReactViewTarget(DependencyObject originalSource, Point point)
+        {
+            // If the target is not a child of the root view, then this pointer
+            // event does not belong to React.
+            if (!RootViewHelper.IsReactSubview(originalSource))
+            {
+                return null;
+            }
+
+            // population of sources provided by:
+            // http://stackoverflow.com/questions/2059475/what-is-the-silverlights-findelementsinhostcoordinates-equivalent-in-wpf
+            var sources = new List<DependencyObject>();
+            //changed from external edits, because VisualHit is
+            //only a DependencyObject and may not be a UIElement
+            //this could cause exceptions or may not be compiling at all
+            //simply filter the result for class UIElement and
+            //cast it to IEnumerable<UIElement> if you need
+            //the very exact same result including type
+
+            VisualTreeHelper.HitTest(
+                _view,
+                null,
+                new HitTestResultCallback(
+                    (HitTestResult hit) => {
+                        sources.Add(hit.VisualHit);
+                        return HitTestResultBehavior.Continue;
```

Comment:
Because the last check returns a UIElement, we can't move it into this block, but we'll move the first two checks.